### ISSUE TITLE
1.11.1 and 1.1.5 seem to overlap

### ISSUE THREAD

**Issue by jmanico** [OWASP/ASVS#1184]
"components"

**Comment by tghosth** [comment]
Agree, need to think about this more...

**Comment by elarlang** [comment]
**V1.1 Secure Software Development Lifecycle**

| # | Description | L1 | L2 | L3 | CWE |
| :---: | :--- | :---: | :---: | :---: | :---: |
| **1.1.5** | Verify definition and security analysis of the application's high-level architecture and all connected remote services. ([C1](https://owasp.org/www-project-proactive-controls/#div-numbering)) | | ✓ | ✓ | 1059 |


**V1.11 Business Logic Architecture**

| # | Description | L1 | L2 | L3 | CWE |
| :---: | :--- | :---: | :---: | :---: | :---: |
| **1.11.1** | Verify the definition and documentation of all application components in terms of the business or security functions they provide. | | ✓ | ✓ | 1059 |

**Comment by tghosth** [comment]
This is what 3.0.1 looked like:
![image](https://user-images.githubusercontent.com/13623828/211584747-8f833e31-2727-43de-8467-214bdd9490a6.png)


I am struggling a little to see the origin and neither of these two requirements are super clear.

**Comment by jmanico** [comment]
> This is what 3.0.1 looked like: ![image](https://user-images.githubusercontent.com/13623828/211584747-8f833e31-2727-43de-8467-214bdd9490a6.png)
> 
> I am struggling a little to see the origin and neither of these two requirements are super clear or actionable.

I share your concern, Josh.

**Comment by tghosth** [comment]
Ok this would be my attempt to clarify 

| # | Description | L1 | L2 | L3 | CWE |
| :---: | :--- | :---: | :---: | :---: | :---: |
| **1.1.5** | **_Verify that a high-level application architecture document exists which also includes remote services upon which the application relies and how interactions with them are secured. ([C1](https://owasp.org/www-project-proactive-controls/#div-numbering))_** | | ✓ | ✓ | 1059 |
| **1.1.5** | Verify definition and security analysis of the application's high-level architecture and all connected remote services. ([C1](https://owasp.org/www-project-proactive-controls/#div-numbering)) | | ✓ | ✓ | 1059 |
| **1.1.6** | Verify that a high-level architecture for the application and all connected remote services has been defined and security has been addressed in that architecture. | | ✓ | ✓ | tbd | tbd |
| **1.3** | A high-level architecture for the application and all connected remote services has been defined and security has been addressed in that architecture. |  | ✓ | ✓ | 1.0 |

**Comment by jmanico** [comment]
I suggest we keep 1.1.2 and drop all of the other requirements that tell me *how* to do threat modeling. 

1.1.2 | Verify the use of threat modeling for every design change or sprint planning to identify threats, plan for countermeasures, facilitate appropriate risk responses, and guide security testing. |   | ✓ | ✓ | 1053
-- | -- | -- | -- | -- | --

specifically I think 1.1.4, 1.1.5, 1.1.6 and similar should all go away.

1.1.4 | Verify documentation and justification of all the application's trust boundaries, components, and significant data flows. |   | ✓ | ✓ | 1059
-- | -- | -- | -- | -- | --
1.1.5 | Verify definition and security analysis of the application's high-level architecture and all connected remote services. (C1) |   | ✓ | ✓ | 1059
1.1.6 | Verify implementation of centralized, simple (economy of design), vetted, secure, and reusable security controls to avoid duplicate, missing, ineffective, or insecure controls. (C10) |   | ✓ | ✓ | 637

1.1.4	Verify documentation and justification of all the application's trust boundaries, components, and significant data flows.		✓	✓	1059
1.1.5	Verify definition and security analysis of the application's high-level architecture and all connected remote services. ([C1](https://owasp.org/www-project-proactive-controls/#div-numbering))		✓	✓	1059
1.1.6	Verify implementation of centralized, simple (economy of design), vetted, secure, and reusable security controls to avoid duplicate, missing, ineffective, or insecure controls. ([C10](https://owasp.org/www-project-proactive-controls/#div-numbering))		✓	✓	637

**Comment by elarlang** [comment]
> I think all documentation requirements should be respectfully deleted.

No. Without documented decisions, it is not possible to develop, configure, and test/verify, whether the application works as expected (as no one knows, what is the "expected")

How we going to use documentation requirements, is up for discussion in https://github.com/OWASP/ASVS/discussions/1831. But the choices there are not just to delete them, but "to have them separately in V1 or as a first chapter in related category".

**Comment by tghosth** [comment]
Agree, we need to keep documentation requirements subject to #1831.

**Comment by tghosth** [comment]
Not also the discussion in #1541

**Comment by elarlang** [comment]
1.11.1 and 1.1.5 are discussed in different issues and the topic here is that they seem to overlap.

I think 1.11.1 and 1.1.5 are clearly separate requirements
* 1.1.5 - document the (external) services the application **need to use**
* 1.11.1 - document the services the application **provides**

So from overlap perspective I think there is no problem to solve and we can close the issue.

ping @jmanico @tghosth

**Comment by jmanico** [comment]
I agree @elarlang I think these are different. But I do suggest a small change to 1.11.1 - just change business to business logic. This is a really small request and you can ignore it so we can move along. Your call.


1.11.1 | Verify the definition and documentation of all application components in terms of the business _logic_ or security functions they provide. |   | ✓ | ✓ | 1059
-- | -- | -- | -- | -- | --

**Comment by elarlang** [comment]
Current requirement 1.11.1:
> Verify the definition and documentation of all application components in terms of the business or security functions they provide.

[Proposal](https://github.com/OWASP/ASVS/issues/1184#issuecomment-1378298598) from Josh:
> Verify that the high-level architecture defines all application components in terms of the business or security functions they provide.

[Proposal](https://github.com/OWASP/ASVS/issues/1184#issuecomment-2413567955) from Jim:
> Verify the definition and documentation of all application components in terms of the business logic or security functions they provide.

I think Josh's proposal was during "early-days" before we clarified the scope for V1. So I prefer clear documentation-oriented requirement.

**Comment by jmanico** [comment]
Agreed, @elarlang

**Comment by tghosth** [comment]
Ok so I agree that there is no overlap and rather we are looking to clarify, based on input from Jim and Elar and the discussion above, I would suggest the following:

| # | Description | L1 | L2 | L3 | CWE |
| :---: | :--- | :---: | :---: | :---: | :---: |
| **1.1.5** | [MODIFIED] Verify that architecture documentation defines the remote services upon which the application relies and how interactions with them are secured. | | ✓ | ✓ | 1059 |
| **1.11.1** |  [MODIFIED] Verify that architecture documentation defines all application components in terms of the business or security functions they provide. | | ✓ | ✓ | 1059 |

Do we want to discuss here @elarlang whether these are crucial requirements from the perspective of "without this, your app is insecure"?

**Comment by jmanico** [comment]
Can we change "business" to "business logic" please? Business logic is a really specific term that helps us point to the work of v11.

**Comment by elarlang** [comment]
> Do we want to discuss here @elarlang whether these are crucial requirements from the perspective of "without this, your app is insecure"?

Here I prefered to solve only the overlap topic.

1.1.5 - is important to list all outgoing connections and services, that is precondition for allowlisting outgoing connections - for that we have a separate issue #993 and this should be discussed there

1.11.1 - for me it does not cover any clear pre-condition value for other requirement and we can just delete it.

**Comment by tghosth** [comment]
So @jmanico even if I edit the requirement to include logic, do you think that this is a crucial requirement from the perspective of "without this, your app is insecure"?

| # | Description | L1 | L2 | L3 | CWE |
| :---: | :--- | :---: | :---: | :---: | :---: |
| **1.11.1** |  [MODIFIED] Verify that architecture documentation defines all application components in terms of the business logic or security functions they provide. | | ✓ | ✓ | 1059 |

**Comment by jmanico** [comment]
> So @jmanico even if I edit the requirement to include logic, do you think that this is a crucial requirement from the perspective of "without this, your app is insecure"?

Other than access control documentation requirements I think that all logging requirement should go away :) So no.

**Comment by elarlang** [comment]
So the proposal is to delete 1.11.1 as out of scope - only documentation without containing a clear security decision as precondition for other requirements.
